src/meta/server_state.cpp

```diff
@@ -3269,8 +3269,373 @@ void server_state::set_max_replica_count(configuration_set_max_replica_count_rpc
+                             update_partition_max_replica_count(partition_index, rpc);
+                         },
+                         server_state::sStateHash,
+                         std::chrono::milliseconds(100));
+        return;
+    }
+
+    dassert_f(context.stage == config_status::not_pending,
+              "invalid config status while updating max_replica_count: context.stage={}, "
+              "app_name={}, app_id={}, partition_index={}, new_max_replica_count={}",
+              enum_to_string(context.stage),
+              app_name,
+              app_id,
+              partition_index,
+              new_max_replica_count);
+
+    context.stage = config_status::pending_remote_sync;
+    context.pending_sync_request.reset();
+    context.msg = nullptr;
+
+    auto new_partition_config = old_partition_config;
+    new_partition_config.max_replica_count = new_max_replica_count;
+    ++(new_partition_config.ballot);
+    context.pending_sync_task =
+        update_partition_max_replica_count_on_remote(new_partition_config, rpc);
+}
+
+// ThreadPool: THREAD_POOL_META_STATE
+task_ptr server_state::update_partition_max_replica_count_on_remote(
+    const partition_configuration &new_partition_config,
+    configuration_set_max_replica_count_rpc rpc)
+{
+    const auto &gpid = new_partition_config.pid;
+    const auto app_id = gpid.get_app_id();
+    const auto partition_index = gpid.get_partition_index();
+    const auto new_max_replica_count = new_partition_config.max_replica_count;
+    const auto new_ballot = new_partition_config.ballot;
+    const auto &app_name = rpc.request().app_name;
+
+    auto level = _meta_svc->get_function_level();
+    if (level <= meta_function_level::fl_blind) {
+        dwarn_f("have to wait until meta level becomes more than fl_blind, then process the "
+                "current request of updating max_replica_count: current_meta_level={}, "
+                "app_name={}, app_id={}, partition_index={}, new_max_replica_count={}, "
+                "new_ballot={}",
+                _meta_function_level_VALUES_TO_NAMES.find(level)->second,
+                app_name,
+                app_id,
+                partition_index,
+                new_max_replica_count,
+                new_ballot);
+
+        // NOTICE: pending_sync_task should be reassigned
+        return tasking::enqueue(
+            LPC_META_STATE_HIGH,
+            tracker(),
+            [this, new_partition_config, rpc]() {
+                const auto &gpid = new_partition_config.pid;
+                const auto partition_index = gpid.get_partition_index();
+
+                const auto &app_name = rpc.request().app_name;
+                auto &response = rpc.response();
+
+                zauto_write_lock l(_lock);
+
+                auto app = get_app_and_check_exist(app_name, response);
+                if (app == nullptr) {
+                    response.hint_message += fmt::format(
```

Comment:
same, use `+=` reason?

Reply:
> same, use `+=` reason?

Actually there are 2 cases that a table does not exist. One is the check at the beginning of setting max_replica_count that once the table does not exist, just returning the error is ok since user might have inputed a wrong table name. 

However, another case is that the table becomes disappeared during the process of updating the max_replica_count. Therefore I append some hints here to tell where this exception has happened.

Reply:
Don' t understand, if one rpc completed,  it fill `hint_message` and return, why exist `append`,  will it happend `two error case` meanwhile